src/Sportradar.OddsFeed.SDK/Entities/REST/Internal/MarketNames/EntityNameExpression.cs

```diff
@@ -107,13 +107,21 @@ private async Task<string> GetHomeCompetitor(CultureInfo culture)
             if (_sportEvent is IMatch match)
             {
                 var competitor = await InvokeAndWrapAsync(match.GetHomeCompetitorAsync).ConfigureAwait(false);
+                await competitor.EnsureProfileLoaded();
```

Comment:
Up to this point all of the code is making use of Async/await so it was safe to await at this point and not need any other code changes.